File: packages/agent-core/src/todo.ts
import type { TodoId, RunId } from "./ids.js";

export type AgentTodoStatus = "pending" | "running" | "done" | "blocked" | "cancelled";

export type AgentTodoCreatorType = "agent" | "user" | "system";

export type AgentTodo = {
  id: TodoId;
  runId: RunId;
  title: string;
  description?: string;
  status: AgentTodoStatus;
  createdBy: AgentTodoCreatorType;
  order: number;
};

File: backend/src/aithru_agent/domain/todo.py
from enum import StrEnum
from typing import Literal

from .base import AithruBaseModel


class AgentTodoStatus(StrEnum):
    PENDING = "pending"
    RUNNING = "running"
    DONE = "done"
    BLOCKED = "blocked"
    CANCELLED = "cancelled"


AgentTodoCreatorType = Literal["agent", "user", "system"]


class AgentTodo(AithruBaseModel):
    id: str
    run_id: str
    title: str
    description: str | None = None
    status: AgentTodoStatus
    created_by: AgentTodoCreatorType
    order: int

